services/github_publisher.py

```
@@ -0,0 +1,320 @@
+                    if response.status_code == 422:
+                        # Branch might already exist, try to get it
+                        response = client.get(
+                            f"{self.base_url}/repos/{owner}/{repo}/git/refs/heads/{feature_branch}",
+                            headers=self.headers
+                        )
+                        if response.status_code != 200:
+                            response.raise_for_status()
+                    else:
+                        response.raise_for_status()
+                        
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)